logcat/src/commonMain/kotlin/logcat/Logcat.kt

```diff
@@ -91,3 +91,71 @@ inline fun logcat(
     }
   }
 }
+
+/**
+ * An alternative way of calling the free [logcat] method using a Timber-like API.
+ *
+ * Use like:
+ *
+ * ```kotlin
+ * import logcat.logcat
+ *
+ * class MouseController {
+ *
+ *   fun play {
+ *     var state = "CHEEZBURGER"
+ *     logcat.d { "I CAN HAZ $state?" }
+ *     // logcat output: D/MouseController: I CAN HAZ CHEEZBURGER?
+ *
+ *     logcat.i { "DID U ASK 4 MOAR INFO?" }
+ *     // logcat output: I/MouseController: DID U ASK 4 MOAR INFO?
+ *
+ *     logcat.w { exception.asLog() }
+ *     // logcat output: W/MouseController: java.lang.RuntimeException: FYLEZ KERUPTED
+ *     //                        at sample.MouseController.play(MouseController.kt:22)
+ *     //                        ...
+ *
+ *     logcat.e("Lolcat") { "OH HI" }
+ *     // logcat output: E/Lolcat: OH HI
+ *   }
+ * }
+ * ```
+ */
+@Suppress("ClassName")
+object logcat {
+  context(subject: Any)
+  fun v(
+    tag: String? = null,
+    message: () -> String
+  ) = subject.logcat(LogPriority.VERBOSE, tag, message)
+
+  context(subject: Any)
+  fun d(
+    tag: String? = null,
+    message: () -> String
+  ) = subject.logcat(DEBUG, tag, message)
+
+  context(subject: Any)
+  fun i(
+    tag: String? = null,
+    message: () -> String
+  ) = subject.logcat(LogPriority.INFO, tag, message)
+
+  context(subject: Any)
+  fun w(
+    tag: String? = null,
+    message: () -> String
+  ) = subject.logcat(LogPriority.WARN, tag, message)
```

Comment:
Since these are not inlined, the tag of the caller won't be captured.

Separately, it's an intentional design decision to not expose N methods

> Most of the time, our developers just want to "send something to logcat" without even thinking about priority. logcat() picks "debug" as the right default to provide more consistency across a large codebase. Making the priority a parameter also means only one method to learn, and you don't have to learn / think about priorities prior to writing a log. This becomes especially important when there are several parameters requiring overloads (e.g. in Timber (6 priorities + 1 generic log method) * 3 overloads = 21 methods to choose from).